File: partiql-ast/src/main/java/org/partiql/ast/v1/expr/ExprParameter.java
package org.partiql.ast.v1.expr;

import org.jetbrains.annotations.NotNull;
import org.partiql.ast.v1.AstNode;
import org.partiql.ast.v1.AstVisitor;

import java.util.Collection;
import java.util.Collections;

/**
 * TODO docs, equals, hashcode
 */
public class ExprParameter extends Expr {
    public int index;

    public ExprParameter(int index) {
        this.index = index;
    }

    @Override
    @NotNull
    public Collection<AstNode> children() {
        return Collections.emptyList();
    }

    @Override
    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
        return visitor.visitExprParameter(this, ctx);
    }
}

File: partiql-ast/src/main/kotlin/org/partiql/ast/v1/expr/ExprParameter.kt
package org.partiql.ast.v1.expr

import org.partiql.ast.v1.AstNode
import org.partiql.ast.v1.AstVisitor

/**
 * TODO docs, equals, hashcode
 */
public class ExprParameter(
    @JvmField
    public var index: Int,
) : Expr() {
    public override fun children(): Collection<AstNode> = emptyList()

    public override fun <R, C> accept(visitor: AstVisitor<R, C>, ctx: C): R =
        visitor.visitExprParameter(this, ctx)
}
